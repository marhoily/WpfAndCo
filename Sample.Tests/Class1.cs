using System.Linq;
using ApprovalTests;
using ApprovalUtilities.Utilities;
using FluentAssertions;
using Microsoft.Data.Entity;
using Microsoft.Data.Entity.Metadata;
using Microsoft.Data.Entity.Metadata.Conventions;
using Microsoft.Data.Entity.Metadata.Conventions.Internal;
using Xunit;

namespace Sample
{
    public class Class1
    {
        [Fact]
        public void FactMethodName()
        {
            var builder = new ModelBuilder(
                new CoreConventionSetBuilder().CreateConventionSet());
            builder.Entity<Person>();
            builder.Entity<City>();
            IModel ro = builder.Model;
            var city = ro.FindEntityType(typeof(City).FullName);
            Approvals.VerifyAll(city.GetProperties()
                .Select(x =>x.WritePropertiesToString()), "");
            var property = city.GetProperties().First();
            property.ClrType.Name.Should().Be("Int64");
            property.Name.Should().Be("Id");
        }
    }
}
