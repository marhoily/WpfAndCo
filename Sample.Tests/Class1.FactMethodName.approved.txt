[0] = IProperty
{
	ClrType: System.Int64
	IsNullable: False
	IsReadOnlyBeforeSave: True
	IsReadOnlyAfterSave: True
	IsStoreGeneratedAlways: False
	ValueGenerated: OnAddOrUpdate
	RequiresValueGenerator: True
	IsShadowProperty: False
	IsConcurrencyToken: False
}

[1] = IProperty
{
	ClrType: System.String
	IsNullable: False
	IsReadOnlyBeforeSave: False
	IsReadOnlyAfterSave: False
	IsStoreGeneratedAlways: False
	ValueGenerated: Never
	RequiresValueGenerator: False
	IsShadowProperty: False
	IsConcurrencyToken: False
}

